lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -93,6 +93,13 @@ export default function FilePickerApp({ onSubmit }) {
             formFields={formFields}
           />
         )}
+        {content && (
+          <input
+            name="document_url"
+            type="hidden"
+            value={content.file.id}
+          />
```

Comment:
The `document_uri` field, and other hidden form fields are now handled by the `FilePickerFormFields` component.

Reply:
Related question: How does the backend know that the `document_url` field contains a file ID as opposed to a URL?

Reply:
I think the frontend may have to do this for us.

When the `document_url` is a URL it needs to look like `https://example.com/foo/`.

When it's a Blackboard file ID it needs to look like `blackboard://content-resource/xyz123`.

I think it's the frontend that nows which type of document the user picked so the frontend is gonna need to format this `blackboard://content-resource/<ID>` string.

See https://stackoverflow.com/c/hypothesis/questions/356

Reply:
Actually, maybe the frontend doesn't need to actually format the `blackboard://` URL. That URL format can remain a backend-internal thing. But the frontend does need to indicate to the backend that this is a Blackboard file ID rather than a URL or something else. For example by submitting a `blackboard_file_id` field (rather than `document_url`)